### PR TITLE
[PBW-6200] - Fixes stuck loading spinner on Edge

### PR DESCRIPTION
**Pull request by jpfranco**
[Link to ticket](https://jira.corp.ooyala.com/browse/PBW-6200)

The problem was caused by the fact that MS Edge doesn't trigger `playing` or `canplaythrough` after seeking ([example](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/6168727/)), so the buffering state was not being reset. It's also worth noting that `canplaythrough` [isn't properly implemented on all browsers](https://bugs.chromium.org/p/chromium/issues/detail?id=73609), so it's probably better not to rely on it too much. 

The `playing` event seemed like a good place to raise the `BUFFERED` event, but according to the spec, it is not necessarily supposed to be fired after seeking. This means that the only way to determine whether the video has resumed after buffering is to check that `currentTime` is getting updated on the `timeupdate` event.

### CODE REVIEW

File: src/main/js/main_html5.js [hunk 1]
```diff
@@ -1016,6 +1017,15 @@ require("../../../html5-common/js/utils/environment.js");
      * @param {object} event The event from the video
      */
     var raiseTimeUpdate = _.bind(function(event) {
+      // [PBW-6200] - MS Edge will not fire 'playing' or 'canplaythrough' events after seeking.
+      // If playback progresses and 'waitingEventRaised' hasn't been cleared, we fire a
+      // 'buffered' event in order to notify that the video has stopped buffering
```
Comment (pilievOoyala) [inline src/main/js/main_html5.js]: should we isolate this code to just edge? Does it mess up the other browsers?

Comment (pilievOoyala) [inline src/main/js/main_html5.js]: And does this report Buffered more than before?

**Comment by aeng7** [comment]
Can we add unit tests for this?

**Comment by jpfranco** [comment]
@aeng7 sure, I'll add some unit tests for this.

File: src/main/js/main_html5.js (hunk 1 above)
Comment (aeng7) [inline src/main/js/main_html5.js]: What happens when you seek again after an initial seek but before the video is able to play before a buffer? Will timeUpdate be triggered in this case and fire a BUFFERED event?

I think if there exists a reliable event to detect when the video starts playback, we should use it if possible. If canplaythrough does not meet this criteria, is there another event we can listen to? If not, timeupdate sounds good.

**Comment by aeng7** [comment]
@jpfranco awesome, thanks!

File: src/main/js/main_html5.js (hunk 1 above)
Comment (jpfranco) [inline src/main/js/main_html5.js]: @aeng7 I need to confirm this, but I believe that the `timeupdate` event will not get fired until seeking is done. So if you seek again while the video is seeking, there won't be any `timeupdate`'s in between.

It does seem to me that `timeupdate` is the reliable way to detect that playback has resumed. The other options are `canplay` and `canplaythrough` which don't indicate that the video is actually playing (even though it usually is when these are fired), only that it can be played. Also, `play` and `playing` are supposed to be fired when playback starts after the video was stopped or paused, so some browsers interpret seeking as a pause, but Edge doesn't.

Comment (aeng7) [inline src/main/js/main_html5.js]: @jpfranco alright, sounds good. Please confirm to see that timeupdates are not sent when seeking again before the video starts playback. Thanks!

**Comment by jpfranco** [comment]
After doing a lot of testing I discovered that by simply including Edge on `startUnderflowWatcher` I had already fixed the issue, so my other changes were redundant. The `startUnderflowWatcher` already has logic that raises a `canplaythrough` event after playback resumes. This inclusion was necessary anyway because Edge doesn't raise the `waiting` event when playback stalls due to buffering. 

I still think that `timeupdate` is a better place to handle buffering than `canplaythrough`, but after reviewing the code I think that changing this would involve a much bigger effort. 

I had included some unit tests, but they no longer apply after these changes. I think all of the scenarios are already covered on `underflow-tests.js`.

**Comment by aeng7** [comment]
The raiseCanPlayThrough function name is a little deceiving, since it raises a BUFFERED message bus event and not a canPlayThrough video event. Great stuff though that this simple fix works. Can you add some Edge specific tests to underflow-tests.js? The tests currently in underflow-tests.js test a good combination of browsers. Now that we are including Edge to the underflow watcher, we should include some tests for Edge as well.

**Comment by jpfranco** [comment]
@aeng7 np, added a unit test. I think this is the only case that wasn't already covered indirectly.

**Comment by jpfranco** [comment]
@aeng7 I've updated the unit tests. Can you take another look please?

File: tests/unit/main_html5/underflow-tests.js [hunk 2]
```diff
@@ -457,6 +458,20 @@ describe('main_html5 chrome underflow tests', function () {
     expect(_.contains(vtc.notified, vtc.interface.EVENTS.WAITING)).to.be(true);
   });
 
+  it('should raise buffered event after waiting on MS Edge when currentTime has progressed', function(){
+    OO.isChrome = false;
```
Comment (mityaha) [inline tests/unit/main_html5/underflow-tests.js]: I recommend to set all OO.is{Browser,Platform} constants to false in afterEach - this way we won't depend on previous test to set to false constants that were used there

Comment (jpfranco) [inline tests/unit/main_html5/underflow-tests.js]: @mityaha I think that this is already handled by the `beforeEach` above. I think that `isChrome` is set to true by default because it is used in most tests, so that's why I need to set it to false here. This test doesn't depend on the previous one to set its values though.

Comment (mityaha) [inline tests/unit/main_html5/underflow-tests.js]: I see...  This goes against the pattern used in other test files, but OK

**Comment by aeng7** [comment]
LGTM!
